src/orange/OrangeInputIO.json.cc

```diff
@@ -538,6 +538,11 @@ void from_json(nlohmann::json const& j, OrangeInput& value)
         }
     }
 
+    if (auto iter = j.find("logic_notation"); iter != j.end())
+    {
+        logic::from_json(*iter, value.logic_notation);
+    }
```

Comment:
For compatibility with existing files, this should have
```suggestion
    }
    else
    {
        value.logic_notation = LogicNotation::infix;
    }
```